### PR TITLE
Replace ServiceMonitors in kyma-system by annotation based scraping

### PR DESCRIPTION
**Pull request by a-thaler**
**Description**
Requires release notes as migration script is needed!

The main motivation is to decouple the components from the monitoring component. It also fits the future strategy of not scraping kyma components via ServiceMonitors anymore.

Changes proposed in this pull request:

- removed `servicemonitors` from all components except `monitoring`
- removed `prometheusrule` -> no Prometheus Operator CRDs are used anymore by kyma components (except monitoring)
- annotated all kyma services which were scraped by a `servicemonitor`
- removed annotations on pod level when needed. As the reconciler will not remove them on upgrade, I disabled the scrape annotation first
- defined a scrape job hardcoded in Prometheus config to discover the service endpoints in `kyma-system` namespace by annotation
- I compared the targets before and after. All are included, additionally, there is a target for the `ory-postgresql`
- fixed `Eventing / NATS Delivery` dashboard, that are using many-to-one vector matching (group_left). Enabling scraping via annotations duplicates the affected timeseries until the service monitors are cleaned up. It means that for some period of time, the abovementioned dashboard remains broken. To fix it, a `max` function is used to duplicate the timeseries.

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/11300

### CODE REVIEW

**Review by kwiatekus — DISMISSED** [review]

**Review by cnvergence — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

File: docs/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+title: Migration Guide
+---
+
+As preparation for the upcoming modularization and having a reduced set of dependencies on a module, Kyma switched to a annotation based scraping of metrics for system components. With that the ServiceMonitors of the components needs to get cleaned up. When you upgrade from Kyma 2.9 to 2.10, either run the script [cleanup.sh](./assets/cleanup.sh) or run the commands from the script manually.
```
Comment (grego952) [inline docs/migration-guide.md:5]:
```suggestion
In preparation for the upcoming modularization and having a reduced set of dependencies on a module, Kyma switched to an annotation-based scraping of metrics for system components. With that, the ServiceMonitors of the components need to get cleaned up. When you upgrade from Kyma 2.9 to 2.10, either run the script [cleanup.sh](./assets/cleanup.sh) or run the commands from the script manually.
```

**Review by grego952 — APPROVED** [review]

**Review by kwiatekus — DISMISSED** [review]

**Review by grego952 — DISMISSED** [review]

**Review by skhalash — DISMISSED** [review]

**Review by kwiatekus — APPROVED** [review]

**Review by triffer — DISMISSED** [review]

**Comment by skhalash** [comment]
Failing eventing upgrade tests uncover the following problem: after a cluster upgrade, the service monitors remain untouched (the reconciler cannot delete resources). That's why certain targets become duplicated, which breaks some PromQL queries. That's the reason why the tests fail. But it also breaks some Grafana dashboards (e.g. `Published Event Types` with a `Subscription and Published Event Types without Subscription` panels in the `Kyma / Eventing / NATS Delivery dashboard`).  Cleaning up the service monitors fixes the problem. However, it will not fix the data in TSDB, so the dashboards remain broken if the selected time window overlaps with the pre-upgrade time.

**Comment by skhalash** [comment]
Will split the PR into isolated small PRs per component

**Pull request closed without merging**